doc/design/python-api-1.md

```diff
@@ -0,0 +1,194 @@
+Designs for an improved Python API
+==================================
+
+* New base classes:
+    * `Dandiset`
+        * `identifier: str`
+        * `created: datetime`
+        * `modified: datetime`
+        * `version: Optional[Version]`
+        * `get_assets() -> Iterator[Asset]`
+        * `get_assets_under_path(path: str) -> Iterator[Asset]` — Returns all assets whose paths are either under the given folder or equal the given filepath
+        * `get_asset_by_path(path: str) -> Optional[Asset]` — Finds the unique asset with the given exact path; returns `None` if no such asset
+        * `get_metadata() -> DandisetMeta`
+   * `Asset`:
+        * `path: str`
+        * `size: int`
+        * `created: datetime`
+        * `modified: datetime`
+            * **Problem:** The API reports `created` and `modified` dates when paginating through a version's assets, but the data is absent when getting an asset's metadata directly from the asset-specific endpoint (Note that, although there are two timestamps for modification times in asset metadata, neither of them match the `modified` time in the pagination results.)
```

Comment:
yeah -- those `created` and `modified` AFAIK are for the specific row in dandiset_versions_assets -- so specific to that dandiset version asset, whenever metadata record modified, should be the one on when metadata record itself was modified (should be before those `created` `modified` above).  If our `Asset` here  is not dandiset/version specific (I think it is not), then I guess we could have only the dateModified from metadata.    For our needs (upload/download `refresh` mode) I think it is the one which is needed. So let's just drop `created` (even if for just now).

Reply:
`created` removed.